### ISSUE TITLE
[Material 2] `AppBar.actions` horizontal padding is not conforming according to the M2 specs

### ISSUE THREAD

**Issue by twoco** [flutter/flutter#118965]
Not actually a bug. Rather an UI issue. Maybe I'm too pedantic. The space of the app bar buttons is not equal. The action buttons are too close on the edge. Or vice versa, the leading button has too much space?

![demo](https://user-images.githubusercontent.com/53160891/213896253-1dd9e1bb-31ee-48e4-995e-42f7ca222672.gif)
![demo](https://user-images.githubusercontent.com/53160891/213896256-0468b543-a263-4643-9ac5-3615b5a5681a.png)

```dart
AppBar(
  title: const Text("Title"),
  titleSpacing: 8,
  // Own leading button to suppress the tooltip.
  // Flutter should disable it by default or offer an option.
  leading: IconButton(
    icon: const Icon(Icons.arrow_back),
    onPressed: () {
      Navigator.of(context).pop();
    },
  ),
  actions: [
    IconButton(
      icon: const Icon(Icons.arrow_forward),
      onPressed: () {},
    ),
  ],
)
```

https://m2.material.io/components/app-bars-top#specs

What do you think? Close it if I'm wrong. I will manually add some padding to compensate for this. We could also offer an option to adjust the space like `titleSpacing`.

**Comment by JesutoniAderibigbe** [comment]
Hey, Twoco!

Please do this

Wrap your actions icon button with Padding. You can use the padding widget and use EdgeInsets.all(8.0)

**Comment by Shatha-Naami** [comment]
Hey Twoco,
In `AppBar` class, there are variable called **leadingWidth**. 
The default value is `leadingWidth = 56;`

So, you can increase or decrease this value to fit your design.
PS: In your case, the good value I think is `leadingWidth = 48;`

**Comment by twoco** [comment]
Thanks, but my issue is more about Material specs in Flutter. The initial official state without custom adjustments. Both sides should have the same space. But to be honest, I'm not sure if the [MD specs](https://m2.material.io/components/app-bars-top#specs) work great here. They are symmetrical, which is good. But in my case, 16px or 24px is a bit too close on the edge.

Btw. the property `leadingWidth` works a bit weird. Because setting it to `32` results in a space of 24px to the icon (same as the right side of action buttons). It also  interferes with `titleSpacing` (overlapping). Maybe I found a bug here?

Workarounds:
- Set `leadingWidth` to 32 (px) to have the same space as on the right side of the actions (24px).
- Wrap the action IconButtons in a container with padding (40 - 24 = 16px).

Maybe the best solution is to ofer a new property like `actionWidth` or `trailingWidth` (pendant of `leadingWidth`). Without changing anything to prevent breaking changes. Or just close this issue if you think it is correct as is.

**Comment by exaby73** [comment]
## Triage report

I can reproduce this issue with `Master (3.7.0-24.0.pre.62)`. From the second screenshot, it looks like the horizontal padding is being sort of squashed. I haven't been able to pinpoint the exact code causing this, but I think this should be an easy fix overall.

<img width="400" src="https://user-images.githubusercontent.com/50555895/213994753-25ae5935-dc54-4d7c-aa09-9add566aec32.png">

![image](https://user-images.githubusercontent.com/50555895/213994456-47d6db68-026e-46e6-b593-3da5ab97ee58.png)

<details><summary>Code Sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
        leading: const BackButton(),
        actions: const [RotatedBox(quarterTurns: 2, child: BackButton())],
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headline4,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }
}
```

</details>

<details><summary><code>flutter doctor -v</code> (Master)</summary>

```
[!] Flutter (Channel master, 3.7.0-24.0.pre.62, on macOS 13.1 22C65 darwin-arm64, locale en-US)
    • Flutter version 3.7.0-24.0.pre.62 on channel master at /Users/nabeelparkar/fvm/versions/master
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/2.18.6/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/nabeelparkar/fvm/versions/master. Consider adding /Users/nabeelparkar/fvm/versions/master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6ae7ad72c1 (3 hours ago), 2023-01-23 00:35:35 -0500
    • Engine revision 92313596d7
    • Dart version 3.0.0 (build 3.0.0-152.0.dev)
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/nabeelparkar/Library/Android/sdk/
    • Platform android-33, build-tools 33.0.0
    • ANDROID_SDK_ROOT = /Users/nabeelparkar/Library/Android/sdk/
    • Java binary at: /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /Applications/Brave Browser.app/Contents/MacOS/Brave Browser

[✓] Android Studio (version 2021.3)
    • Android Studio at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.1)
    • IntelliJ at /Users/nabeelparkar/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 71.3.6
    • Dart plugin version 223.8214.16

[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.1)
    • IntelliJ at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/223.8214.52/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.56.0

[✓] Connected device (3 available)
    • iPhone 14 Pro Max (mobile) • 6BC61E88-4DE0-4BCB-8599-DA5294269146 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-2 (simulator)
    • macOS (desktop)            • macos                                • darwin-arm64   • macOS 13.1 22C65 darwin-arm64
    • Chrome (web)               • chrome                               • web-javascript • Brave Browser 109.1.47.171

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>

**Comment by exaby73** [comment]
@twoco There seems to be a duplicate issue already (https://github.com/flutter/flutter/issues/91884). I am closing this issue in favor of that one. Thank you
